Sources/Xcodeproj/Plist.swift

```diff
@@ -0,0 +1,50 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+/// A enum representing data types for legacy Plist type.
+/// see: https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/PropertyLists/OldStylePlists/OldStylePLists.html
+enum Plist {
+    case string(String)
+    case array([Plist])
+    case dictionary([String: Plist])
+}
+
+extension Plist: ExpressibleByStringLiteral {
+    public typealias UnicodeScalarLiteralType = StringLiteralType
+    public typealias ExtendedGraphemeClusterLiteralType = StringLiteralType
+
+    public init(unicodeScalarLiteral value: UnicodeScalarLiteralType) {
+        self = .string(value)
+    }
+    public init(extendedGraphemeClusterLiteral value: ExtendedGraphemeClusterLiteralType) {
+        self = .string(value)
+    }
+    public init(stringLiteral value: StringLiteralType) {
+        self = .string(value)
+    }
+}
+
+private let escapedQuote = "\\\""
+
+extension Plist {
+    /// Serializes the Plist enum to string.
+    func serialize() -> String {
+        switch self {
+        case .string(let str):
+            // If the string has spaces Xcode represents it as: `"\"some string\""`.
+            let hasSpace = str.utf8.contains(UInt8(ascii: " "))
+            return "\"" + (hasSpace ? escapedQuote : "") + str.escape(ascii: "\"") + (hasSpace ? escapedQuote : "") + "\""
```

Comment:
Since we're only writing and not reading them, I think a great simplification would be to always quote them.  So for every string you would just start with a double-quote, then append the string with any `"` and `\` characters prefixed with `\`, and then another double-quote.  That's all you need to do here.

Build setting quoting will need to be handled at a higher level.